### PR TITLE
Feature/Keen Tracking Preprint Views and Downloads [EOSF-347]

### PR DESCRIPTION
**Pull request by pattisdr**
## Ticket
https://openscience.atlassian.net/browse/EOSF-347

❗️ Now blocked by discover-page component PR https://github.com/CenterForOpenScience/ember-osf/pull/169
❗️ For Devops - need to setup system variables for Keen private and public projects, process.env.PREPRINTS_PRIVATE_PROJECT_ID, process.env.PREPRINTS_PRIVATE_WRITE_KEY, process.env.PREPRINTS_PUBLIC_PROJECT_ID, process.env.PREPRINTS_PUBLIC_WRITE_KEY
❗️ TODO - REMOVE PINNED EMBER-OSF COMMIT ONCE DISCOVER-PAGE COMPONENT MERGED.
❗️ Download Preprint button not working (issue outside of this PR, seems to be switch to fileDownloadURL.content)
## Purpose

Tracks pageviews of preprints in Keen.  Tracks download counts of preprint file and supplementary files for noncontributors only. Tracks views of preprint file and supplementary files for all users.  Because we're using `ember-metrics`, and ember-osf now has an ember-metrics Keen adapter, events that were being sent just to GA, should start automatically being sent to Keen as well.

## Changes
When primary or supplementary file downloaded, these are logged under the **non-contributor-preprint-downloads** collection.  Sent to both public and private preprints projects, so can be retrieved for later display. File downloads are tracked for non-contribs only. Selected data:
```
 "download_info": {
        "preprint": {
            "type": "preprint",
            "id": "fcrax"
        },
        "file": {
            "primaryFile": true,
            "id": "58763e8963ca2f4790fa8336",
            "version": 1
        }
    },
    "interaction": {
        "action": "click",
        "category": "link",
        "url": "http://localhost:7777/v1/resources/t5hyf/providers/osfstorage/58763e8963ca2f4790fa8336",
        "label": "Content - Download as Non-Contributor"
    },
```
When supplementary or primary preprint files are viewed, these are logged under the **preprint-file-views** collection.   When the content page is loaded along with the primary file that counts as a view of the primary file.  When the user toggles through supplementary files, then those are logged as views of supplementary files. Sent to both public and private preprints projects, so can be retrieved for later display. Selected data: 
```
 "file_views": {
        "preprint": {
            "type": "preprint",
            "id": "fcrax"
        },
        "file": {
            "primaryFile": true,
            "id": "58763e8963ca2f4790fa8336",
            "version": 1
        }
    }
```

Tracking **pageviews** of content page.  See pageViewed key to see which page was visited.  Timestamps and other information are also sent alongside the pageViewed key. Selected section of keen data:
```
"pageViewed": {
        "page": "/7epdw/",
        "title": "content"
    }
```

## Notes for Reviewers
Default for ember-metrics is that these events will only fire in production. To fire events locally, you should have GA and/or Keen accounts created and put ID's in your config. Under metricsAdapters, change environments to ['all'].
```
metricsAdapters: [
            {
                name: 'GoogleAnalytics',
                 environments: ['all'], <--- Change to this
                config: {
                    id: process.env.PREPRINTS_GOOGLE_ANALYTICS_ID
                }
            },
            {
                name: 'Keen',
                environments: ['all'], <--- Change to this
                config: {
                    private: {
                        projectId: process.env.PREPRINTS_PRIVATE_PROJECT_ID,
                        writeKey: process.env.PREPRINTS_PRIVATE_WRITE_KEY
                    },
                    public: {
                        projectId: process.env.PREPRINTS_PUBLIC_PROJECT_ID,
                        writeKey: process.env.PREPRINTS_PUBLIC_WRITE_KEY
                    }
                }
            }
        ],
```
## Side effects

Events tracked in GA should automatically start being tracked in Keen as well.

### CODE REVIEW

**Comment by binoculars** [comment]
Do we need ember-browserify in this repo? I'm don't see it used in the diff, so it might be a leftover from the refactor, but it would best to keep the dependencies to a minimum if possible

**Comment by pattisdr** [comment]
It seemed like I needed ember-browserify to load some of the JS modules, like keen-tracking needed in the Ember-osf Keen metrics adapter, even though ember-browserify is installed in ember-osf.  https://github.com/ef4/ember-browserify#using-ember-browserify-in-addons

File: config/environment.js [hunk 1]
```diff
@@ -117,11 +117,26 @@ module.exports = function(environment) {
...
                     id: process.env.GOOGLE_ANALYTICS_ID
                 }
-            }
+            },
+            {
+                name: 'Keen',
+                environments: ['production'],
+                config: {
+                    private: {
+                        projectId: 'process.env.PREPRINTS_PRIVATE_PROJECT_ID',
```
Comment (binoculars) [inline config/environment.js]: These shouldn't be wrapped in quotes, because they'll get evaluated as literal strings.

Comment (pattisdr) [inline config/environment.js]: Oops, thanks.

**Comment by brianjgeiger** [comment]
@pattisdr Merge conflicts. Please move back to Passed CR once the tests pass if everything seems simple enough.

**Comment by brianjgeiger** [comment]
Could you resolve conflicts, please?

**Comment by brianjgeiger** [comment]
(I promise I'll get it merged right after this time. :) )

**Comment by pattisdr** [comment]
@brianjgeiger, I think we should wait on this. It is now dependent on the new discover-page component.

**Comment by brianjgeiger** [comment]
Will the new discover page component handle all of this functionality, or will we still need this but updated to include extra things after the new component is in?

**Comment by pattisdr** [comment]
The discover-page component does **not** handle all of this functionality and we still need it. However, the discover-component is making some changes to analytics that improves this PR.

**Comment by brianjgeiger** [comment]
When you have a moment, merge conflicts

**Comment by pattisdr** [comment]
Conflicts resolved @brianjgeiger.

**Comment by brianjgeiger** [comment]
Sorry, Merge conflicts again. I'll merge quickly once they're resolved.

**Comment by pattisdr** [comment]
Looking at failure..

**Comment by pattisdr** [comment]
conflicts resolved @brianjgeiger

**Pull request merged**